### ISSUE TITLE
Uploading new documents via interface💡 

### ISSUE THREAD

**Issue by bw-Deejee** [vemonet/libre-chat#1]
### Describe the feature

It would be nice to be able to upload new documents in the web interface. Uploaded documents would have to be ingested by the model and be accessable in the current session.

**Comment by vemonet** [comment]
Hi @bw-Deejee, thanks a lot for the nice suggestion!

I just added it to the API, it can be done easily through the OpenAPI swagger UI, with a user friendly file picker. Files will be automatically uploaded to the `documents_path`, if you provide `.zip` files they will be automatically unzipped, finally the vectorstore will be rebuild, so that the changes are directly taken into account by the running chatbot

Since it is implemented through the API it can also be done programmatically, e.g. having a script that feeds your QA chatbot with documents

It is live on the `main` branch and docker `main` tag. I'll publish soon a new release

Note that I recently made some changes to the volume setup in the `docker-compose.yml`, so you might want to check the docs to update your setup if you are using docker

**Comment by bw-Deejee** [comment]
Hi @vemonet, thanks for the quick reply and for this cool project.

I'm not sure if I misunderstood your answer. I cloned the newest version of the repo just now and ran the "hatch run dev" but I don't see an option to upload files right in the web interface yet.
Is this a thing you will implement in the future or is it already there and I have to configure something differently?

**Comment by vemonet** [comment]
No, the upload UI is not directly in the "chat UI", but in the "API  UI" which you can access by clicking the green icon top right

The chat UI is really optimized for humans

The API UI is a web page automatically generated to help using the different API calls. It stays quite human friendly, while also enabling programmatic access 

Now that you say this I am thinking I should make this more clear maybe :)

**Comment by bw-Deejee** [comment]
Ok, completely understood now. I found the document upload and tested it.
Coming back to this feature request though, I think it would be really nice to have this upload feature right at the chat UI or at least on a seperate page which is as user friendly.
For my purpose this is a real central feature of this tool and this is why I would love to see it being integrated as such. I do see your point though with the programmatic access approach.
I might fiddle around with it a bit and integrate it on the chat UI myself but I'm sure you would do a way better job at it 😄

**Comment by emojized** [comment]
I like the software very much... going full docker is in my opinion not so cool. I am happy its running without docker. but still jahs soem issues. For example I cant run the faiss vectorstore , it creates it and says "killed" out of nowehere.
